### PR TITLE
Fix errors

### PR DESCRIPTION
**Pull request by mattn**
I'm using Go tip and got error.
(Also fixes another error when return empty lines from format)

### CODE REVIEW

File: autoload/go/mod.vim [hunk 1]
```diff
@@ -8,7 +8,11 @@ function! go#mod#Format() abort
   " go mod only exists in `v1.11`
   if empty(s:go_major_version)
     let tokens = matchlist(go#util#Exec(['go', 'version']), '\d\+.\(\d\+\)\(\.\d\+\)\? ')
-    let s:go_major_version = str2nr(tokens[1])
+    if len(tokens) > 0
+      let s:go_major_version = str2nr(tokens[1])
+    else
+      let s:go_major_version = 13
```
Comment (mattn) [inline autoload/go/mod.vim]: Do you prefer 0 and ignore checking below?

**Comment by bhcleek** [comment]
Thanks @mattn.

What are `go version` and `go list` outputting?

**Comment by mattn** [comment]
```
C:\>go version
go version devel +1ab063ce53 Fri May 17 22:32:30 2019 +0000 windows/amd64
```

And `go list` outside GOPATH should be an error.

**Comment by bhcleek** [comment]
`go list` outside `GOPATH` with `*.go` files present and without a go.mod file (i.e. a null module) returns the absolute path with a `_` prefix. With no `*.go` files present `go list` already returns a non-zero exit code and `go#package#FromPath` returns -1 in that case on line 163 (https://github.com/fatih/vim-go/blob/f45093914a9dcfa8d39d5849c100b0118c310fcd/autoload/go/package.vim#L162-L164).

Can you provide more information about your use case for the change to `go#package#FromPath`?

**Comment by mattn** [comment]
Trying outside GOPATH.

terminal7

**Comment by bhcleek** [comment]
Thanks! So to be clear, you're solving for the case of trying to use completion of packages when in a null module?

**Comment by mattn** [comment]
Yes, it is.

File: autoload/go/package.vim [hunk 2]
```diff
@@ -82,13 +86,21 @@ function! s:vendordirs() abort
...
     let [l:dir, l:err] = go#util#ExecInDir(['go', 'list', '-f', '{{.Dir}}'])
     if l:err != 0
       return []
     endif
-    let l:dir = split(l:dir, '\n')[0]
+    let l:lines = split(l:dir, '\n')
+    if empty(l:lines)
+      return {}
+    endif
+    let l:dir = l:lines[0]
```
Comment (bhcleek) [inline autoload/go/package.vim]: Why is this second new block necessary? Assuming that `go list` in a null modules exits 0, is there some condition where `.Root` will have a value, but `.Dir` will not?

File: autoload/go/package.vim [hunk 3]
```diff
@@ -163,7 +175,11 @@ function! go#package#FromPath(arg) abort
     return -1
   endif
 
-  let l:importpath = split(l:out, '\n')[0]
+  let l:lines = split(l:out, '\n')
+  if empty(l:lines)
+    return -1
+  endif
+  let l:importpath = l:lines[0]
```
Comment (bhcleek) [inline autoload/go/package.vim]: How can `go list` with no arguments exit 0 _and_ not have any lines in its output?

File: autoload/go/package.vim (hunk 2 above)
Comment (mattn) [reply autoload/go/package.vim]: No. go list return `.` not null.

image

Comment (mattn) [reply autoload/go/package.vim]: go list possibly return empty and 0.

image

Comment (mattn) [reply autoload/go/package.vim]: Oops. Sorry. This is my fault.

**Comment by mattn** [comment]
Sorry, This is my mistake. Ignore this.

**Comment by bhcleek** [comment]
I think there was some good stuff in here. Are you sure you want to close it?

**Comment by mattn** [comment]
Ah, checking tip is required.

**Comment by mattn** [comment]
Rebased from master branch & removed my some changes about empty outputs.

**Comment by bhcleek** [comment]
Thanks, @mattn

**Pull request merged**